### PR TITLE
Update zipdownload.php

### PR DESCRIPTION
**Pull request by lucasmarin**
Adding a counter before .(extension) to files with same name on zip.
Before only the last file on message with equals names was added on zip

### CODE REVIEW

File: plugins/zipdownload/zipdownload.php [hunk 1]
```diff
@@ -141,18 +143,8 @@ public function download_attachments()
...
-
-            if ($filename === null || $filename === '') {
-                $ext      = (array) rcube_mime::get_mime_extensions($part->mimetype);
-                $ext      = array_shift($ext);
-                $filename = $rcmail->gettext('messagepart') . ' ' . $pid;
-                if ($ext) {
-                    $filename .= '.' . $ext;
-                }
-            }
+            $disp_name = $this->_create_displayname($part->filename);
```
Comment (alecpl) [inline plugins/zipdownload/zipdownload.php]: _create_displayname() uses different arguments. Did you test this code?

File: plugins/zipdownload/zipdownload.php [hunk 2]
```diff
@@ -190,6 +182,49 @@ public function download_messages()
...
+    
+    /**
+     * Create and get display name of attachment part to add on zip file
+     *
+     * @param $part stdClass Part of attachment on message
+     * @param $pid string Part Id of attachment
+     *
+     * @return string Display name of attachment part
+     */
+    private function _create_displayname($part, $pid)
```
Comment (alecpl) [inline plugins/zipdownload/zipdownload.php]: $pid is just $part->mime_id, so we don't need an additional argument.

**Review by alecpl — CHANGES_REQUESTED** [review]

File: plugins/zipdownload/zipdownload.php (hunk 1 above)
Comment (lucasmarin) [inline plugins/zipdownload/zipdownload.php]: Ouch. Sorry for mistake, I tested on different version of roundcube (1.1.9), and it confused me to make a commit to a master version... 

I will make a new pull request with correct code and tested with master version

File: plugins/zipdownload/zipdownload.php (hunk 2 above)
Comment (lucasmarin) [inline plugins/zipdownload/zipdownload.php]: The $part variable is updated on

`        foreach ($message->attachments as $part) {
            $pid      = $part->mime_id;
            $part     = $message->mime_parts[$pid];
`
And on the actual code the variable is mentioned directly 

`-            $filename = $part->filename;
 -
 -            if ($filename === null || $filename === '') {
 -                $ext      = (array) rcube_mime::get_mime_extensions($part->mimetype);
 -                $ext      = array_shift($ext);
 -                $filename = $rcmail->gettext('messagepart') . ' ' . $pid;
 -                if ($ext) {
 -                    $filename .= '.' . $ext;
 -                }
 -            }
`
I didn't know that it was intentional to get the right value, or, with some circunstances 
on $part variable updated the value of mime_id wans't set, so I put it with argument on method.

I will update the pull request with this change.

Thanks alecpl.

Regards.

**Comment by lucasmarin** [comment]
Correcting script to merge with master.

Thx @alecpl.

**Pull request merged**